### PR TITLE
Fixes #1859

### LINKED ISSUE #1859: bigip_asm_policy_fetch exported file is limited to 1 MB when using file mode.
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

bigip_asm_policy_fetch

##### ANSIBLE VERSION

```
ansible 2.9.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible']
  ansible python module location = /opt/ansible-2.9.2/lib64/python3.6/site-packages/ansible
  executable location = /opt/ansible-2.9.2/bin/ansible
  python version = 3.6.3 (default, Mar 20 2018, 13:50:41) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### PYTHON VERSION

```
Python 3.6.3
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     14.1.2.6
  Build       0.0.2
  Edition     Point Release 6
  Date        Fri Jun  5 11:36:46 PDT 2020
```

##### OS / ENVIRONMENT

N/A

##### SUMMARY

When using bigip_asm_policy_fetch without the inline option, the resulting policy output is limited to 1 MB. The ASM policy size is around 13 MB.

##### STEPS TO REPRODUCE

```
- name: "Export Policy Inline (this captures the full policy"
  bigip_asm_policy_fetch:
    provider: "{{ provider }}"
    partition: "{{ policy_partition }}"
    name: "{{ policy_name }}"
    inline: yes
  register: policyOutput
  delegate_to: localhost

- name: "Export Policy File Mode (this is limited to 1 MB)"
  bigip_asm_policy_fetch:
    provider: "{{ provider }}"
    partition: "{{ policy_partition }}"
    name: "{{ policy_name }}"
    file: "{{ policy_name }}.xml"
    dest: /var/tmp/asmexport
  delegate_to: localhost
```

##### EXPECTED RESULTS

Expected that the file mode and inline mode results in the entire policy being retrieved.

##### ACTUAL RESULTS

```
$ ls -althr /var/tmp/asmexport/
-rw-r--r--   1 xxxxxxx default_linux_group 1.0M Mar 26 15:25 policy_1
-rw-r--r--   1 xxxxxxx default_linux_group 1.0M Mar 26 15:26 policy_2
-rw-r--r--   1 xxxxxxx default_linux_group 1.0M Mar 26 15:27 policy_3
-rw-r--r--   1 xxxxxxx default_linux_group 1.0M Mar 26 15:28 policy_4
```

### PR DESCRIPTION
**Pull request by nitinthewiz**
Fixes the issue where ASM file download is restricted to the first MB as it wasn't chunking the download.

### CODE REVIEW

File: ansible_collections/f5networks/f5_modules/plugins/module_utils/icontrol.py [hunk 1]
```diff
@@ -293,43 +293,57 @@ def __exit__(self, exc_type, exc_value, exc_tb):
...
+                break
+            start += chunk_size
+            if start >= size:
+                break
+            if (end + chunk_size) > size:
+                end = size - 1
             else:
-                error = "Invalid Content-Length value returned: %s ," \
-                        "the value should be greater than 0" % length
-                raise F5ModuleError(error)
```
Comment (wojtek0806) [inline ansible_collections/f5networks/f5_modules/plugins/module_utils/icontrol.py:332]: Is this bug in ASM fixed it used to be a major pain in the back, if not needed all good

**Review by wojtek0806 — COMMENTED** [review]
Answer the question i asked otherwise looks good

**Review by wojtek0806 — CHANGES_REQUESTED** [review]
missing changelog fragment

File: ansible_collections/f5networks/f5_modules/plugins/module_utils/icontrol.py (hunk 1 above)
Comment (nitinthewiz) [reply ansible_collections/f5networks/f5_modules/plugins/module_utils/icontrol.py:332]: Oh... Hmmm... I don't know about it being fixed... I'll toss this check back in :)

**Comment by nitinthewiz** [comment]
> missing changelog fragment

added!

**Review by wojtek0806 — APPROVED** [review]
lgtm

**Pull request merged**